docs/source/error_code_list2.rst

```diff
@@ -214,3 +214,20 @@ mypy generates an error if it thinks that an expression is redundant.
 
         # Error: If condition in comprehension is always true  [redundant-expr]
         [i for i in range(x) if isinstance(i, int)]
+
+
+Check that expression is not implicitly true in boolean context [truthy-bool]
+--------------------------------------------------------------------------------
+
+Warn when an expression whose type does not implement __bool__ or __len__ is used in boolean context
+since it could always evaluate to true (if one also assumes that no subtype implements __bool__ or
```

Comment:
"You should always take an umbrella since it could always rain" -- not exactly what we mean :)

Maybe something like "since by default objects evaluate as true"?

The thing with subtypes is correct but I wonder if we can position it differently? Essentially we're warning due to lack of type-based **guarantee** of meaningful bool. What would we say if the python data model would be defined in such a way that `len(object()) == 0)`? What would our warning, if any, be like?